Build failed ''_tzcnt_u32': identifier not found'- with Win10, CUDA v8.0, VS10

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2010
-->

- OpenCV => 3.1 (commit d1bbc0b6e6c0c5115e54aa6d9f32a1d332a9538c -  Thu Nov 3 19:24:52)
- Operating System / Platform => Windows 10 64 Bit
- Compiler => Visual Studio 2010
- CUDA toolkit v8.0
##### Detailed description

Hi! 

I have the following git version of OpenCV - d1bbc0b6e6c0c5115e54aa6d9f32a1d332a9538c -

During a 64bit CUDA enable installation, the build failed with the following error:

1>..\..\..\modules\imgproc\src\contours.cpp(58): error C3861: '_tzcnt_u32': identifier not found

Thanks in advance.

<!-- your description -->

<!-- ##### Steps to reproduce

<!-- to add code example fence it with triple backticks and optional file extension
    ```.cpp
    // C++ code example
    ```
 or attach as .txt or .zip file
-->

It also fails:
`.\..\..\modules\imgproc\src\contours.cpp(56): error C2660: '_BitScanForward' : function does not take 1 arguments`

The function definition redirects to this:
c:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include\intrin.h:
` _BitScanForward(unsigned long* Index, unsigned long Mask)`

Thanks ;)

--- a/modules/imgproc/src/contours.cpp
+++ b/modules/imgproc/src/contours.cpp
@@ -50,10 +50,15 @@
 static const CvPoint icvCodeDeltas[8] =
     { CvPoint(1, 0), CvPoint(1, -1), CvPoint(0, -1), CvPoint(-1, -1), CvPoint(-1, 0), CvPoint(-1, 1), CvPoint(0, 1), CvPoint(1, 1) };
 
+#if CV_SSE2
+static
 inline unsigned int trailingZeros(unsigned int value) {
+    CV_DbgAssert(value != 0); // undefined for zero input (https://en.wikipedia.org/wiki/Find_first_set)
 #if defined(_MSC_VER)
-#if (_MSC_VER < 1500)
-    return _BitScanForward(value);
+#if (_MSC_VER < 1700)
+    unsigned long index = 0;
+    _BitScanForward(&index, value);
+    return (unsigned int)index;
 #else
     return _tzcnt_u32(value);
 #endif
@@ -70,6 +75,7 @@ inline unsigned int trailingZeros(unsigned int value) {
     return MultiplyDeBruijnBitPosition[((uint32_t)((value & -value) * 0x077CB531U)) >> 27];
 #endif
 }
+#endif
 
 CV_IMPL void
 cvStartReadChainPoints( CvChain * chain, CvChainPtReader * reader )

